### PR TITLE
Fix app_flutter flaky tests by updating test golden images

### PR DESCRIPTION
**Pull request by keyonghan**
This is to fix: https://github.com/flutter/flutter/issues/58833

### CODE REVIEW

File: test_utilities/bin/flutter_test_runner.sh [hunk 1]
```diff
@@ -24,7 +24,7 @@ pushd $1 > /dev/null
 flutter packages get
 flutter analyze
 flutter format --line-length=120 --set-exit-if-changed lib/ test/
-flutter test --test-randomize-ordering-seed=random
+flutter test --test-randomize-ordering-seed=random --update-goldens
```
Comment (CaseyHillers) [inline test_utilities/bin/flutter_test_runner.sh:27]: Changes made here should also be made in the windows script for consistency

**Comment by CaseyHillers** [comment]
This shouldn't be run in CI. `--update-goldens` is expected to be run on the developers machine to include the diffs of the app in their PR. If the test is an issue, it should have skip: true. This is hiding any changes to app_flutter look in CI.

**Comment by godofredoc** [comment]
That specific test is flaking very frequently even when there are no changes to flutter-app. Is there a way to set a threshold on the diff? Also when the tests got added they broke local runs on macs can we also skip golden tests if running outside of CI?

**Comment by CaseyHillers** [comment]
(1) There's no way to set a diff range as it's hard coded in the framework
(2) We should make golden testing specific to a platform to prevent any crossplatform issues (different platforms actually render pixels differently 😢 , we face this in app_flutter commits that added the goldens). I filed https://github.com/flutter/flutter/issues/60720 to set it to only run on linux. Can you post an example of the mac failures in that issue?
(3) Can you clarify what you mean by skip golden tests if outside of CI?

**Comment by godofredoc** [comment]
3) I used to run tests locally using test_utilities/bin/global_test_runner.dart but they do not work anymore because I do most of the cocoon development on my laptop.

**Comment by CaseyHillers** [comment]
I think that's related to 2. Before running goldens, we should verify the
platform is linux. I don't think that is possible to pass in the CLI, so
we'll need a Cocoon specific wrapper around matchesGolden that checks the
platform.

If there were non-golden issues on your laptop, please let me know so I can
look into it.

**Pull request merged**